packages/typespec-ts/src/index.ts

```diff
@@ -140,13 +235,18 @@ export async function $onEmit(context: EmitContext) {
   }
 }
 
-function removeUnusedInterfaces(project: Project) {
-  const allInterfaces = project
-    .getSourceFiles()
-    .flatMap((file) => file.getInterfaces());
+/**
+ * Removing this for now, as it has some problem when we have two models with the same name and only one of them is unused, this function will end up removing the other used models.
+ */
+export function removeUnusedInterfaces(project: Project) {
```

Comment:
This is for removing unused models. it fixes the bug that previous the same model name in two sub clients, and one is unused, the other one is being referred, and this function will end up remove both of them. 
  It also fixes that bug when we try to use rename to resolve the conflicts in the `src/models/index.ts`, the top level `src/index.ts` will export those after rename models which has been removed.